src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -607,32 +607,17 @@ void network::allocate_primitives() {
                     return (lhs->get_output_layout().bytes_count() > rhs->get_output_layout().bytes_count());
               });
 
+    // Move layers that can be optimized to the back of nodes_to_allocate
+    std::stable_partition(nodes_to_allocate.begin(),
+                          nodes_to_allocate.end(),
+                          [&po](std::shared_ptr<program_node> const& e) {
+                            return !e->can_be_optimized();
+                          });
+
```

Comment:
I think this can be a good solution. but wasn't it resolved with more simpler solution like  adding another condition on the original sort above? 

`if (lhs->get_output_layout().bytes_count() > rhs->get_output_layout().bytes_count()
        return true;
 else if (lhs->get_output_layout().bytes_count() == rhs->get_output_layout().bytes_count()
        return (lhs->get_unique_id() < rhs->get_unique_id());
  else return false;
`
I think unique id returns the exec_order and it will perform same as your change without adding additional loop.

Reply:
I am concerned about the corner cases that occur when adding a new condition to the sort, and in fact the current conditions seem sufficient to handle the layers to be optimized out. However, I want to make clear that they are created after instantiating other layers. And there will be no performance overhead with vector partitioning.

Reply:
I think still the partitioning solution has potential corener cases when the optimzed layers are cascased, like optimized_reorder->optimized_reshape, and optimized_reshape comes earlier than optimized reorder..

Reply:
Despite they are partitioned, the optimized layers which are cascaded will be instantiated sequentially because they maintain the order as the ordering has sorted before.

Reply:
partitioning happens after sort, so the order might be already sorted by the previous sort, isnt it?

Reply:
Yes, it is. So, I use stable_partition to maintain the sorting order.

Reply:
Yeah so I meant that sorting_order might be exec_order, thereby the original problem still might exist in the cascased optimized out nodes : ) 

For your concern of the side effect, 
having an additional condition for the suggested == condition as follows will resolve your concern, because the memory reuse algorithm will not be called for optimized nodes the other allocating order will be kept same as previous. 

if (lhs->get_output_layout().bytes_count() > rhs->get_output_layout().bytes_count() 
       return true; 
else if ((lhs->get_output_layout().bytes_count() == rhs->get_output_layout().bytes_count()
            && (lhs->can_be_optimized() || rhs->can_be_optimized()) ) 
         return (lhs->get_unique_id() < rhs->get_unique_id()); 
else return false;

So far was my opinion and its up to you which to pick!
Because, actually such cascaded optimized nodes are unpreferrable corner cases and I think they are also able to be handled by other optimization passes. But just we cannot sure they never happens :D